src/treeView/registerTreeView.ts

```diff
@@ -1,31 +1,38 @@
-import { commands, ExtensionContext, window } from "vscode";
-import { getHubStructure } from "../binary/requests/hubStructure";
-import { TABNINE_TREE_NAVIGATION_COMMAND } from "../globals/consts";
+import { commands, ExtensionContext, window, env, Uri } from "vscode";
+import { fireEvent } from "../binary/requests/requests";
+import {
+  TABNINE_APP_URL,
+  TABNINE_OPEN_APP_COMMAND,
+  TABNINE_TREE_NAVIGATION_COMMAND,
+} from "../globals/consts";
 import navigate from "./navigate";
 import TabnineTreeProvider from "./TabnineTreeProvider";
 
-export default async function registerTreeView(
-  context: ExtensionContext
-): Promise<void> {
+export default function registerTreeView(context: ExtensionContext): void {
   try {
-    const structure = await getHubStructure();
-
-    if (structure) {
-      void commands.executeCommand(
-        "setContext",
-        "tabnine.tabnine-navigation-ready",
-        true
-      );
-      context.subscriptions.push(
-        window.registerTreeDataProvider(
-          "tabnine-home",
-          new TabnineTreeProvider()
-        ),
-        commands.registerCommand(TABNINE_TREE_NAVIGATION_COMMAND, (view) => {
-          void navigate(view);
-        })
-      );
-    }
+    context.subscriptions.push(
+      window.registerTreeDataProvider(
+        "tabnine-home",
+        new TabnineTreeProvider()
+      ),
+      commands.registerCommand(TABNINE_TREE_NAVIGATION_COMMAND, (view) => {
+        void navigate(view);
+        void fireEvent({
+          name: "hub-opened-from-sidebar",
+        });
+      }),
+      commands.registerCommand(TABNINE_OPEN_APP_COMMAND, () => {
+        void env.openExternal(Uri.parse(TABNINE_APP_URL));
```

Comment:
maybe it would be better to route it through the binary ? by default vscode will inform the user that it is about to open a url (via a modal popup)

Reply:
ping